iptables/iptables.go

```diff
@@ -0,0 +1,104 @@
+package iptables
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/log"
+
+	"github.com/Azure/azure-container-networking/platform"
+)
+
+// cni iptable chains
+const (
+	CNIInputChain  = "AZURECNIINPUT"
+	CNIOutputChain = "AZURECNIOUTPUT"
+)
+
+// standard iptable chains
+const (
+	Input       = "INPUT"
+	Output      = "OUTPUT"
+	Forward     = "FORWARD"
+	Prerouting  = "PREROUTING"
+	Postrouting = "POSTROUTING"
+)
+
+// Standard Table names
+const (
+	Filter = "filter"
+	Nat    = "nat"
+)
+
+// target
+const (
+	Accept     = "ACCEPT"
+	Drop       = "DROP"
+	Masquerade = "MASQUERADE"
+)
+
+// actions
+const (
+	Insert = "I"
+	Append = "A"
+	Delete = "D"
+)
+
+func ChainExists(tableName, chainName string) bool {
+	cmd := fmt.Sprintf("iptables -w 60 -t %s -L %s", tableName, chainName)
+	if _, err := platform.ExecuteCommand(cmd); err != nil {
+		return false
+	}
+
+	return true
+}
+
+func CreateCNIChain(tableName, chainName string) error {
+	var err error
+
+	if !ChainExists(tableName, chainName) {
+		cmd := fmt.Sprintf("iptables -w 60 -t %s -N %s", tableName, chainName)
+		_, err = platform.ExecuteCommand(cmd)
+	} else {
+		log.Printf("%s Chain exists in table %s", chainName, tableName)
+	}
+
+	return err
+}
+
+func RuleExists(tableName, chainName, match, target string) bool {
+	cmd := fmt.Sprintf("iptables -w 60 -t %s -C %s %s -j %s", tableName, chainName, match, target)
+	_, err := platform.ExecuteCommand(cmd)
+	if err != nil {
+		return false
+	}
+
+	return true
+}
+
+func InsertIptableRule(tableName, chainName, match, target string) error {
+	if RuleExists(tableName, chainName, match, target) {
+		log.Printf("Rule already exists")
+		return nil
+	}
+
+	cmd := fmt.Sprintf("iptables -w 60 -t %s -I %s 1 %s -j %s", tableName, chainName, match, target)
```

Comment:
I will recommend defining a constant for iptables command? It seems like you are using it at multiple places.

Reply:
will do